File: objectiveai-sdk-py/objectiveai_sdk/vector/completions/cache/http.py
"""HTTP functions for vector completion cache endpoints."""

from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from objectiveai.client import ObjectiveAI
    from objectiveai.vector.completions.cache import (
        CacheVote,
        CacheVoteRequest,
        CompletionVotes,
        GetCompletionVotesRequest,
    )


async def get_completion_votes(
    client: ObjectiveAI, params: GetCompletionVotesRequest,
) -> CompletionVotes:
    """Retrieve votes for a specific vector completion."""
    return await client.post_unary("vector/completions/votes", params)


async def get_cache_vote(
    client: ObjectiveAI, params: CacheVoteRequest,
) -> CacheVote:
    """Retrieve a cached vote."""
    return await client.post_unary("vector/completions/cache", params)

File: objectiveai-sdk-js/src/vector/completions/cache/http.ts
import { ObjectiveAI, type RequestOptions } from "../../../client";
import type { VectorCompletionsCacheCompletionVotes } from "./completionVotes";
import type { VectorCompletionsCacheCacheVote } from "./cacheVote";
import type { VectorCompletionsCacheCacheVoteRequest } from "./cacheVoteRequest";

export function vectorCompletionsCacheGetCompletionVotes(
  client: ObjectiveAI,
  id: string,
  options?: RequestOptions,
): Promise<VectorCompletionsCacheCompletionVotes> {
  return client.post_unary<VectorCompletionsCacheCompletionVotes>(
    "vector/completions/votes",
    { id },
    options,
  );
}

export function vectorCompletionsCacheGetCacheVote(
  client: ObjectiveAI,
  body: VectorCompletionsCacheCacheVoteRequest,
  options?: RequestOptions,
): Promise<VectorCompletionsCacheCacheVote> {
  return client.post_unary<VectorCompletionsCacheCacheVote>(
    "vector/completions/cache",
    body,
    options,
  );
}
